examples/refrigerator-app/silabs/src/EventHandlerLibShell.cpp

```diff
@@ -0,0 +1,178 @@
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // Convert the argument to an integer
+    int value = std::stoi(argv[0]); // Safe to use now, as we validated the input earlier
+
+    RefrigeratorAlarmEventData * data = Platform::New<RefrigeratorAlarmEventData>();
+    data->eventId                     = Events::Notify::Id;
+    data->doorState                   = static_cast<AlarmBitmap>(atoi(argv[0]));
+
+    DeviceLayer::PlatformMgr().ScheduleWork(EventWorkerFunction, reinterpret_cast<intptr_t>(data));
+
+    return CHIP_NO_ERROR;
+}
+
+/**
+ * @brief configures Refrigerator matter shell
+ *
+ */
+
+CHIP_ERROR RegisterRefrigeratorEvents()
+{
+    static const shell_command_t sRefrigeratorSubCommands[] = {
+        { &RefrigeratorHelpHandler, "help", "Usage: refrigeratoralarm <subcommand>" },
+        { &EventRefrigeratorCommandHandler, "event", " Usage: refrigeratoralarm event <subcommand>" }
+    };
+
+    static const shell_command_t sRefrigeratorEventSubCommands[] = {
+        { &EventHelpHandler, "help", "Usage : refrigeratoralarm event <subcommand>" },
+        { &RefrigeratorDoorEventHandler, "door-state-change", "Sends door state change event to Refrigerator app" }
+    };
+
+    static const shell_command_t sRefrigeratorEventAlarmDoorSubCommands[] = {
+        { &AlarmHelpHandler, "help", "Usage : Refrigerator event to change door state" }
+    };
+
+    static const shell_command_t sRefrigeratorCommand = { &RefrigeratorCommandHandler, "refrigeratoralarm",
+                                                          "refrigerator alarm commands. Usage: refrigeratoralarm <subcommand>" };
+
+    sShellRefrigeratorEventAlarmDoorSubCommands.RegisterCommands(sRefrigeratorEventAlarmDoorSubCommands,
+                                                                 ArraySize(sRefrigeratorEventAlarmDoorSubCommands));
+    sShellRefrigeratorEventSubCommands.RegisterCommands(sRefrigeratorEventSubCommands, ArraySize(sRefrigeratorEventSubCommands));
+    sShellRefrigeratorSubCommands.RegisterCommands(sRefrigeratorSubCommands, ArraySize(sRefrigeratorSubCommands));
+
+    Engine::Root().RegisterCommands(&sRefrigeratorCommand, 1);
+
+    return CHIP_NO_ERROR;
+}
+
+void EventWorkerFunction(intptr_t context)
+{
+    VerifyOrReturn(reinterpret_cast<void *>(context) != nullptr, ChipLogError(Shell, "EventWorkerFunction - Invalid work data"));
+    EventData * data = reinterpret_cast<EventData *>(context);
+
+    switch (data->eventId)
+    {
+    case Events::Notify::Id: {
+        RefrigeratorAlarmEventData * alarmData = reinterpret_cast<RefrigeratorAlarmEventData *>(context);
+        RefrigeratorAlarmServer::Instance().SetStateValue(kRefEndpointId, alarmData->doorState);
+        break;
+    }
+
+    default: {
+        ChipLogError(Zcl, "Invalid Event Id %s, line %d", __func__, __LINE__);
+        break;
+    }
+        // Free memory
+        Platform::Delete(data);
+    }
```

Comment:
this shouldn't be in the switch { } well maybe it works but I never seen it done this way
```suggestion
    }
    
    // Free memory
    Platform::Delete(data);
```

Reply:
added this change